howso/hierarchy.amlg

```diff
@@ -939,4 +942,46 @@
 		))
 	)
 
+	;checks if the env variable HOWSO_ENGINE_LIMITED_FS is set to true (not case sensitive), ignore passed in 'filepath' parameter
+	#!ValidateFilepathParameter
+	(seq
+		(declare (assoc
+			env_vars
+				(last (system "system"
+					;pull all environment variables, 'set' in windows, 'env' in posix
+					(if (= "Windows" (system "os")) "set" "env" )
+				))
+			ignore_filepath_parameter (false)
+		))
+
+		;convert the one string that contains all the variables into a list of pairs of [env_var, value]
+		(assign (assoc
+			env_vars
+				(map
+					;split "var=value" by that first =
+					(lambda (split (current_value) "=" 1))
+					;split into individual env variable strings
+					(split env_vars "\n")
+				)
+		))
+
+		(map
+			(lambda
+				;if the env variable HOWSO_ENGINE_LIMITED_FS is set to true (not case sensitive), ignore filepath parameter
+				(if (and
+						(= "HOWSO_ENGINE_LIMITED_FS" (first (current_value)))
+						(contains_value (last (current_value)) "[tT][rR][uU][eE]" )
+					)
+					(conclude
+						(assign (assoc ignore_filepath_parameter (true) ))
```

Comment:
Is there any reason to not just directly assign over `filepath` here in the `(map)`? rather than ticking the flag and doing it just below?